### PR TITLE
Update the VSTS task so it supports Kubernetes. Note that only SSH co…

### PR DESCRIPTION
**Pull request by PeterJausovec**
…nnection is currently supported, still need to work on the direct connection.

### CODE REVIEW

File: src/tasks/dockerDeploy/dockerDeploy.ts [hunk 1]
```diff
@@ -9,7 +9,8 @@ tl.cd(tl.getInput("cwd"));
 var targetType = tl.getInput("targetType", true);
 /* tslint:disable:no-var-requires */
 require({
-    "ACS DCOS": "./dockerDeployAcsDcos"
+    "ACS DCOS": "./dockerDeployAcsDcos",
+    "Kubernetes": "./dockerDeployAcsKube"
```
Comment (stepro) [inline src/tasks/dockerDeploy/dockerDeploy.ts]: If this is specific to Kubernetes on ACS, we should call it "ACS Kubernetes"

File: src/tasks/dockerDeploy/task.json [hunk 2]
```diff
@@ -36,7 +36,8 @@
             "label": "Target Type",
             "required": true,
             "options": {
-                "ACS DCOS": "Azure Container Service (DC/OS)"
+                "ACS DCOS": "Azure Container Service (DC/OS)",
+                "Kubernetes": "Kubernetes Cluster"
```
Comment (stepro) [inline src/tasks/dockerDeploy/task.json]: "Azure Container Service (Kubernetes)"

File: src/tasks/dockerDeploy/task.json [hunk 3]
```diff
@@ -45,7 +46,6 @@
             "type": "filePath",
             "label": "Docker Compose File",
             "required": true,
-            "visibleRule": "targetType = ACS DCOS",
```
Comment (stepro) [inline src/tasks/dockerDeploy/task.json]: Please keep the visible Rule and OR in include targetType = ACS Kubernetes, as in:

"visibleRule": "targetType = ACS DCOS | targetType = ACS Kubernetes"

File: src/tasks/dockerDeploy/task.json [hunk 4]
```diff
@@ -99,6 +103,18 @@
             "helpMarkDown": "Select an Azure Container Service DC/OS connection type."
         },
         {
+            "name": "kubernetesEndpointType",
+            "type": "pickList",
+            "label": "Kubernetes Connection Type",
+            "required": true,
+            "options": {
+                "Direct": "Direct",
```
Comment (stepro) [inline src/tasks/dockerDeploy/task.json]: If Direct isn't supported yet, don't expose it in the UI.

File: src/tasks/dockerDeploy/task.json (hunk 3 above)
Comment (stepro) [inline src/tasks/dockerDeploy/task.json]: Make sure to fix this in all places.

File: src/tasks/dockerDeploy/task.json (hunk 2 above)
Comment (stepro) [inline src/tasks/dockerDeploy/task.json]: Also the key should be "ACS Kubernetes"

File: src/tasks/dockerDeploy/task.json (hunk 3 above)
Comment (PeterJausovec) [inline src/tasks/dockerDeploy/task.json]: Fixed

File: src/tasks/dockerDeploy/task.json [hunk 5]
```diff
@@ -145,8 +140,8 @@
             "type": "connectedService:ssh",
             "label": "Kubernetes SSH Connection",
             "required": true,
-            "visibleRule": "targetType = Kubernetes && kubernetesEndpointType = SSH",
-            "helpMarkDown": "Select a Kubernetes SSH connection."
+            "visibleRule": "targetType = ACS Kubernetes && kubernetesEndpointType = SSH",
+            "helpMarkDown": "Select an AzurE container Service Kubernetes SSH connection."
```
Comment (stepro) [inline src/tasks/dockerDeploy/task.json]: Typo: AzurE => Azure

**Pull request merged**